examples/lang/erc721/src/lib.rs

```diff
@@ -0,0 +1,230 @@
+// Copyright 2018-2019 Parity Technologies (UK) Ltd.
+// This file is part of ink!.
+//
+// ink! is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// ink! is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with ink!.  If not, see <http://www.gnu.org/licenses/>.
+
+#![cfg_attr(not(feature = "std"), no_std)]
+
+use ink_core::{
+    env::DefaultSrmlTypes,
+    memory::format,
+    storage,
+};
+use ink_lang::contract;
+use ink_model::EnvHandler;
+use ink_core::storage::alloc::Initialize as _;
+use ink_core::storage::alloc::AllocateUsing as _;
+use std::ops::Index;
+
+pub type Result<T, E> = core::result::Result<T, E>;
+
+contract! {
+    #![env = DefaultSrmlTypes]
+
+    // Event deposited when a token transfer occurs
+    event Transfer {
+        from: AccountId,
+        to: AccountId,
+        token_id: u32,
```

Comment:
I think it would be better to create a type alias `pub type TokenId = u32;` instead of using `u32` directly.

Reply:
Also then you can simply name the field `id` because it is clear what kind of id this refers to.